client/src/main/scala/org/http4s/client/PoolManager.scala

```diff
@@ -74,10 +74,11 @@ private final class PoolManager[F[_], A <: Connection[F]](
   private def numConnectionsCheckHolds(key: RequestKey): Boolean =
     curTotal < maxTotal && allocated.getOrElse(key, 0) < maxConnectionsPerRequestKey(key)
 
-  private def isExpired(k: RequestKey, t: Instant): Boolean =
-    waitExpiryTime(k).isFinite() && t
-      .plus(waitExpiryTime(k).toSeconds, ChronoUnit.SECONDS)
-      .isBefore(Instant.now())
+  private def isExpired(t: Instant): Boolean = {
+    val elapsed = Instant.now().getEpochSecond - t.getEpochSecond
```

Comment:
Why is this in seconds instead of milliseconds?  Is there a performance boost for the loss of granularity?

Reply:
Because the timeouts are specified in seconds (aren't they?).

Reply:
Do they have to be?  I've had subsecond SLAs before.